File: app/lib/src/models/measurement.dart
import 'package:app/src/utils/date_converter.dart';

class Measurement {
  Measurement({
    this.current,
    this.power,
    this.timestamp,
    this.voltage,
  });

  double current;
  double power;
  double timestamp;
  double voltage;

  factory Measurement.fromJson(Map<String, dynamic> json) => Measurement(
        current: json["current"].toDouble(),
        power: json["power"].toDouble(),
        timestamp: json["timestamp"].toDouble(),
        voltage: json["voltage"].toDouble(),
      );

  DateTime get utcDatetime {
    return DateConverter.toUTCfromTimestamp(timestamp);
  }
}

File: device_simulator/src/models/measurement.py
class Measurement():

    def __init__(self, voltage, current, timestamp):
        self.voltage = voltage
        self.current = current        
        self.power = voltage * current
        self.timestamp = timestamp

    def to_dict(self):
        return {
            'voltage': self.voltage,
            'current': self.current,
            'power': self.power,
            'timestamp': self.timestamp
        }
